src/server/Middleware.re

```diff
@@ -3,6 +3,8 @@ type ctx('a) = {
     (~status: Httpaf.Status.t, ~headers: list((string, string))=?, string) =>
     unit,
   req: Httpaf.Request.t,
+  /* TODO: change this to a `unit => option(string)` once middlewares are async */
+  body_string: option(string),
```

Comment:
Actually, we could just change this to be a `unit => option(string)` function now. I think it'd be cleaner and it also allows us to implement it lazily later without changing how you use it.

```reason
type ctx = {
   body: unit => option(string)
};
```